### PR TITLE
Report api

### PR DESCRIPTION
**Pull request by MallanagoudaB**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: module/reports/helper.js [hunk 1]
```diff
@@ -63,11 +63,11 @@ module.exports = class ReportsHelper {
 						tasks: {
 							$elemMatch: {
 								isDeleted: { $ne: true },
-								// syncedAt: { $gte: new Date(startFrom), $lte: new Date(endOf) },
```
Comment (VISHNUDAS-tunerlabs) [inline module/reports/helper.js:66]: @MallanagoudaB why this is removed

File: module/reports/helper.js [hunk 2]
```diff
@@ -101,63 +102,7 @@ module.exports = class ReportsHelper {
 				projectReport[CONSTANTS.common.SUBMITTED_STATUS] = 0
 				projectReport[CONSTANTS.common.INPROGRESS_STATUS] = 0
 				projectReport[CONSTANTS.common.STARTED] = 0
-
-				let types = await this.types()
```
Comment (VISHNUDAS-tunerlabs) [inline module/reports/helper.js:105]: @MallanagoudaB why these codes are removed?

File: module/reports/helper.js [hunk 3]
```diff
@@ -101,63 +102,7 @@ module.exports = class ReportsHelper {
...
-				// 		dataAvailable: false,
-				// 		data: {
-				// 			categories: categories,
-				// 			tasks: tasksReport,
-				// 			projects: projectReport,
-				// 		},
-				// 	},
-				// })
-
-				if (getPdf == true) {
```
Comment (VISHNUDAS-tunerlabs) [inline module/reports/helper.js:125]: @MallanagoudaB what happen if we need pdf?

File: module/reports/helper.js [hunk 4]
```diff
@@ -170,27 +115,16 @@ module.exports = class ReportsHelper {
...
-								} else if (categories[category.name]) {
-									categories[category.name] = categories[category.name] + 1
-								} else {
-									if (category.externalId !== '') {
-										categories[category.externalId] = 1
-									} else {
-										categories[category.name] = 1
-									}
-								}
+								let key = category.externalId || category.name
```
Comment (VISHNUDAS-tunerlabs) [inline module/reports/helper.js:123]: @MallanagoudaB why this logic change. what happens if else part of old logic come in place?

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
reviewed on June 1

File: module/reports/helper.js (hunk 2 above)
Comment (MallanagoudaB) [reply module/reports/helper.js:105]: Because its not using anywhere below

File: module/reports/helper.js (hunk 3 above)
Comment (MallanagoudaB) [reply module/reports/helper.js:125]: Its generating below line no 195 the logic is same

File: module/reports/helper.js [hunk 5]
```diff
@@ -101,63 +103,7 @@ module.exports = class ReportsHelper {
...
-						})
-					} else {
-						return resolve({
-							message: CONSTANTS.apiResponses.REPORTS_DATA_NOT_FOUND,
-							data: [],
-							success: false,
-						})
-					}
-				} else {
+				if (!projectDetails.length > 0) {
```
Comment (VISHNUDAS-tunerlabs) [inline module/reports/helper.js:106]: @MallanagoudaB can you please tell me why we removed this Part [line-128 to 185] ref this :
https://github.com/ELEVATE-Project/project-service/pull/23/files#diff-13ab73c9e43ba8f09171958cd34a636841985118315ae882a151e900cd17a009

When enabling report that also needs to be enabled right

File: module/reports/helper.js [hunk 6]
```diff
@@ -101,63 +103,7 @@ module.exports = class ReportsHelper {
...
 					return resolve({
```
Comment (VISHNUDAS-tunerlabs) [inline module/reports/helper.js:107]: @MallanagoudaB what happen if project data is not there and getPdf is true

File: module/reports/helper.js (hunk 3 above)
Comment (VISHNUDAS-tunerlabs) [reply module/reports/helper.js:125]: @MallanagoudaB I think here we had two type reports entity type reports and project report. In this PR https://github.com/ELEVATE-Project/project-service/pull/23/files#diff-13ab73c9e43ba8f09171958cd34a636841985118315ae882a151e900cd17a009

You completely ignored the entity report . Hence the entire logic was removed I was not able catch it during the review. Is there any particular reason why we removed that?

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
reviewed on june 5 2:52

File: module/reports/helper.js [hunk 7]
```diff
@@ -108,70 +108,72 @@ module.exports = class ReportsHelper {
 						return type.label
 					}
 				})
-				// if (!projectDetails.length > 0) {
-				//Initially not supporting pdf report, hence resolving response here. Remove resolve if pdf capability needs to be added
```
Comment (VISHNUDAS-tunerlabs) [inline module/reports/helper.js:112]: @MallanagoudaB we can remove this commented parts

File: module/reports/helper.js [hunk 8]
```diff
@@ -251,27 +253,28 @@ module.exports = class ReportsHelper {
...
-				// 	delete projectReport[CONSTANTS.common.STARTED]
+				// if ( UTILS.revertStatusorNot(appVersion) ) {
+
+				//     projectReport[CONSTANTS.common.COMPLETED_STATUS] = projectReport[CONSTANTS.common.SUBMITTED_STATUS];
+				//     projectReport[CONSTANTS.common.NOT_STARTED_STATUS] = projectReport[CONSTANTS.common.STARTED];
+				//     delete projectReport[CONSTANTS.common.SUBMITTED_STATUS];
+				//     delete projectReport[CONSTANTS.common.STARTED];
 				// }
 
 				//Initially not supporting pdf report, hence resolving response here. Remove resolve if pdf capability needs to be added
```
Comment (VISHNUDAS-tunerlabs) [inline module/reports/helper.js:264]: @MallanagoudaB this commented codes will be ok to remove

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]

**Pull request merged**